src/main/java/com/example/springboot/DAO/BookDAO.java

```diff
@@ -0,0 +1,45 @@
+package com.example.springboot.DAO;
+
+import java.util.Collection;
+
+import com.example.springboot.Base.Book;
+
+/**
+ * This interface defines a methods for work with database table - "Book"
+ * @author Dmytro Kohut
+ * @version 1.0
+ */
+public interface BookDAO {
```

Comment:
You can create one interface for book and Reader for this operations, but use a templates

Reply:
or my God... One interface for all services. WTF??? Are you kidding me?